antarest/study/service.py

```diff
@@ -181,6 +182,14 @@ def get_disk_usage(path: t.Union[str, Path]) -> int:
     return total_size
 
 
+class TaskProgressRecorder(ICommandListener):
+    def __init__(self, notifier: ITaskNotifier) -> None:
+        self.notifier = notifier
+
+    def notify_progress(self, progress: int) -> None:
+        return self.notifier.notify_progress(progress)
```

Comment:
Note that in the future, we should probably think of a way of adapting the progress of each command to a global progress of the task (think snapshot generation with X commands).

Reply:
Indeed. We'll refactor this when we'll have to as it's only few lines of code for the moment